packages/app/android/src/reactnative/java/io/invertase/firebase/common/SharedUtils.java

```diff
@@ -130,6 +130,23 @@ public static boolean isAppInForeground(Context context) {
     List<ActivityManager.RunningAppProcessInfo> appProcesses = activityManager.getRunningAppProcesses();
     if (appProcesses == null) return false;
 
+    // Check if current activity is a background activity
+    if(json.contains("background_activity_names"))
+    {
+      ArrayList<String> backgroundActivities = json.getArrayValue("background_activity_names");
+
+      if(backgroundActivities.size() != 0) {
+        List<ActivityManager.AppTask> taskInfo = activityManager.getAppTasks();
```

Comment:
This is an API21 call and we are still minSdkVersion 16 https://developer.android.com/reference/android/app/ActivityManager#getAppTasks()

Is there a way to accomplish this using APIs available on 16? If not it's safe to say the audience for API<21 is vanishingly small (RN0.64 bumps minSdkVersion to 21) so I'd accept an SDK version check to gate the API usage and make it safe, and then you would likely need to put a "NewApi" suppression in there to pass the lint check I just installed to guard against this #5208 after one slipped through recently

Reply:
I have added support for API<21